### DISCUSSION TITLE
Agent Status Unknown

### DISCUSSION THREAD

**Discussion by Dercni** [glpi-project/glpi-agent#742] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

Yes, I know

### Describe the bug

Then agent can communicate with the GLPI server
The GLPI server fails to communicate with the agent and continually reports the agent status as unknown.

The GLPI versions is 10.0.15
The agent versions is 1.10

From the GLPI server I can ping the agent
On the GLPI server I can view the agent web GUI @ port 62354

I have set set the agent logging level to 2 and see nothing in the logs that indicates a problem.

I have installed the GLPI-Inventory plugin

The agent correctly reports it's tasks to the GLPI server, which confirms communications.
![image](https://github.com/user-attachments/assets/ce96c955-bb63-446d-950f-dc4dfa147f86)

As I have spent weeks on this now I am reporting it as a bug as I cannot see where I have gone wrong.

### To reproduce

Navigate to the computer inventory information page
Click either the agent status refresh icon or the request inventory refresh icon

### Expected behavior

The GLPI server is able to connect to the agent when either the "Agent status" refresh icon is clicked or the "Request Inventory" refresh icon is clicked. Presently:
Agent status = Unknown
Request inventory = Unknown

### Operating system

Windows

### GLPI Agent version

v1.10

### GLPI version

10.0.15

### GLPIInventory plugin or other plugin version

GLPI Inventory v1.3.5

**Comment by g-bougard** [comment]
Hi @Dercni 

you're asking for support not reporting a glpi-agent bug so I'll move your issue as Q&A question.

Your problem can be a network problem: GLPI needs to have a direct network access to the agent port. Also you should tell if you configured a proxy in GLPI. If this is the case, the proxy probably can't contact agents.

**Comment by Dercni** [comment]
My GLPI server has direct access to the agent port. I can use curl on the GLPI server to return the agent GUI.
The agent is on a remote network and behind a 1:1 NAT
There is no proxy involved.

**Comment by killerwanee** [comment]
Hello, I have a similar issue and didn't find an ansewer
Agent work fine for everything but in the gui, when I click on my computer the 2 elements below are "unkown" (picture below). Clicking doesn't change anything and it's been weeks. Is it a bug, or is my config lacking something ?

Maybe it's related to network openings we need to do ? Is it possible to have a map/schema that show what to open in what direction.

Thanks

![image](https://github.com/user-attachments/assets/3c842d05-0971-43c3-bfe0-1d7608b9352c)

**Reply by g-bougard** [reply]
Hello,

from glpi to glpi-agent, by default, you only need to open TCP port 62354 to be able to reach agent status and being able to request inventory. Firewall on the computer where glpi-agent runs should also not block incoming connections on that port. And finally, the computer must not have a masquerade ip address from glpi point of view.

**Reply by killerwanee** [reply]
Thank you very much.
It resolved my problem. There was no rooting from my server to my clients. It was one-sided from the clients to the server.
I can add for other people that could have the issue that for the "request inventory" to work correctly, I needed to add a trusted ip (my glpi server ip) during installation with HTTPD_TRUST as explained in the documentation "installation>windows>command line parameters".

**Reply by g-bougard** [reply]
Hi @killerwanee 

great.

In theory, you don't have to set httpd-trust if configured server url hostname can be resolved to the expected ip as glpi-agent integrates this ip by default as trusted ip. But you may have to set it if the resolved ip is not the same. This depends on your web server setup.
